### DISCUSSION TITLE
Is there any way to open a database from an &[u8]?

### DISCUSSION THREAD

**Discussion by hutchisr** [rusqlite/rusqlite#1105] [Questions]
I'm working on a project where I get the contents of an sqlite3 database in the form of a pointer to a sequence of bytes. Is there any way I can open this as a database in rusqlite? I see `open_in_memory` but that seems to only create a new in-memory database, I don't see a way to specify opening from some already existing memory.

**Comment by gwenn** [comment]
Is it an address / `sqlite3` pointer ? => [from_handle](https://docs.rs/rusqlite/0.24.2/rusqlite/struct.Connection.html#method.from_handle)
Or the database content ? => [Deserialize a database](https://github.com/rusqlite/rusqlite/pull/786)

**Comment by hutchisr** [comment]
The latter, thanks. Looks like it's still a work in progress.

**Comment by TjeuKayim** [comment]
Hi @hutchisr , I'm the author of that pull request to add this feature of opening a database from `&[u8]`. My fork is functional, but it is still waiting for a third review. You're welcome to give it a try and provide feedback.
